### DISCUSSION TITLE
Adding new local storage location (Windows, Docker WSL2)

### DISCUSSION THREAD

**Discussion by guiltlab** [AzuraCast/AzuraCast#5150] [Q&A]
Hello everyone,

First off, thanks a lot to the devs for this amazing, free piece of software. It's really great that this exists.

I am running Azuracast in Docker on a Windows 10 machine. I followed the general Windows install instructions, and set up stations that seem to work perfectly so far. I am having issues however in adding a new local storage location. I seem to only be allowed to add folders located in /var and subfolders, anything else fails. Whenever I try to add a folder that is not located in /var, I get this message:

`ValidationException at /var/azuracast/www/src/Exception/ValidationException.php L37: Object(App\Entity\StorageLocation): Storage location Local: /mnt/d/radio could not be validated: Unable to create a directory at /mnt/d/radio. mkdir(): Permission denied `

In this case I am trying to mount a folder located on an external disk (D:\radio in Windows) but it is the same problem with any folder except those in /var/*  

It appears to be an issue with permissions, but I have no idea how to fix it, and whether the problem is in Windows, WSL, Docker or all of them. I did a lot of reading, and I think I might need to create a new docker volume? but I am not very familiar with Linux commands and this is my first time using Docker so I don't really know what I'm doing. I tried adding a volume with /mnt/d/radio path in docker-compose.yml but that did not do the trick. Starting Docker in admin mode before attempting to add a storage location did not change anything either. The folder I am trying to add as a storage location has been shared with "everyone" with "full control" permissions in Windows. Sadly that did not fix the problem either.

Does anyone have experience with WSL2+Docker+Azuracast by any chance? Thanks in advance!!

**Comment by BusterNeece** [comment]
@fnfzlznddzn I personally use WSL2+Docker to do all of my AzuraCast dev work, but crucially I only mount directories that are also within the Linux filesystem due to the permission issues that can happen when mounting any drives managed by Windows itself.

For Linux permissions issues, we do have an `AZURACAST_PUID` and `AZURACAST_PGID` setting in `.env` (more info here: https://docs.azuracast.com/en/getting-started/settings#the-env-file ) where you can change the UID/GID that AzuraCast runs internally to match whatever your external UID/GID is. Changing this value may allow you access to the Windows shares, but I'm not sure of that.

**Reply by guiltlab** [reply]
Thank you for your answer and your hard work on Azuracast!

Since Windows does not seem to have a concept of UID/GID from my searches, I assume you meant the WSL distro as the "host". I have tried to run `id -u` and `id -g` on Ubuntu in WSL, and got `1000` as a result for both, which seems to match the default values in the `.env` files. So this does not seem to be the issue here :(

**Reply by BusterNeece** [reply]
Yeah, that's what I worried... I'm not sure how to force permissions on Windows mounts inside WSL2. Surely somewhere there are ways of doing it, I just haven't done it myself.

**Reply by guiltlab** [reply]
Thanks for your response and candor. Hopefully someone comes around with a solution! My space on the system disk is limited so this is a pretty frustrating problem. I would rather not run Linux directly to use Azuracast since I use Windows software (in particular foobar2000) daily...

**Reply by BusterNeece** [reply]
I understand, though to be clear AzuraCast is meant to be server-side software. It's typically installed in a datacenter VPS and is built to run as Linux software as a result. A Windows version would be a huge amount of development work in completely unfamiliar territory for us, unfortunately.

**Comment by guiltlab** [comment]
So I found how to do it! Unsurprisingly, the answer was in the Documentation... I just failed to read it properly.
go to https://docs.azuracast.com/en/administration/docker, specifically this paragraph:  [¶](https://docs.azuracast.com/en/administration/docker#mounting-a-directory-into-a-station)
Mounting a directory into a station

services:
  web:
    volumes:
      - /path/on/host/computer:/var/azuracast/stations/<STATION_NAME>/media/

  stations:
    volumes:
      - /path/on/host/computer:/var/azuracast/stations/<STATION_NAME>/media/

I replaced /path/on/host/computer with /mnt/d/radio and <STATION_NAME> with the corresponding name, and my folder was mounted and recognized by Auracast. If I add files to this folder, they get picked up and processed by Azuracast. Awesome!

**Reply by BusterNeece** [reply]
@fnfzlznddzn Ohhhh, I had assumed you had done that part already and it was a permissions issue. Glad you were able to figure it out!

**Reply by xewl** [reply]
And `/mnt/{drive}/path/to/windows/folder´ has worked fine in my case :)
Though it's slower, but for media it's been working out just fine.
